### PR TITLE
Add Wayland LmP Distro

### PR DESCRIPTION
**Pull request by EmbeddedAndroid**
This series adds a new distro configuration to support Wayland/Weston with Foundries.io branding.

### CODE REVIEW

File: meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Weston with Xwayland support (requires X11 and Wayland)
+PACKAGECONFIG[xwayland] = "-Dxwayland=true,-Dxwayland=false,libxcursor"
+
+PACKAGECONFIG_remove_apalis-imx6 = "kms"
+
+# Clients support
+SIMPLE_CLIENTS = "all"
+SIMPLE_CLIENTS_apalis-imx6 = "damage,im,egl,shm,touch,dmabuf-v4l"
```
Comment (mike-scott) [inline meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend:8]: Might need a comment here describing why we're using a smaller subset of simple-clients for apalis-imx6

File: meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Weston with Xwayland support (requires X11 and Wayland)
+PACKAGECONFIG[xwayland] = "-Dxwayland=true,-Dxwayland=false,libxcursor"
+
+PACKAGECONFIG_remove_apalis-imx6 = "kms"
```
Comment (mike-scott) [inline meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend:4]: Might need a comment here describing why we're dropping kms for apalis-imx6.

**Review by mike-scott — CHANGES_REQUESTED** [review]

File: meta-lmp-base/conf/distro/lmp-wayland.conf [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require conf/distro/lmp.conf
+
+DISTRO = "lmp-wayland"
+DISTROOVERRIDES = "lmp:lmp-wayland"
+DISTRO_NAME = "Linux-microPlatform Wayland"
+
+DISTRO_FEATURES_append = " wayland opengl"
+
+#IMX8
+PREFERRED_VERSION_weston_mx8mm = "9.0.0"
```
Comment (mike-scott) [inline meta-lmp-base/conf/distro/lmp-wayland.conf:10]: We can actually move these 2 changes into meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc under the imx8mm section.

**Review by mike-scott — CHANGES_REQUESTED** [review]

File: meta-lmp-base/conf/distro/lmp-wayland.conf (hunk 3 above)
Comment (EmbeddedAndroid) [reply meta-lmp-base/conf/distro/lmp-wayland.conf:10]: Ok will do.

File: meta-lmp-base/recipes-graphics/wayland/weston-init/weston.ini [hunk 4]
```diff
@@ -0,0 +1,72 @@
+# configuration file for Weston
+
+[core]
+#modules=xwayland.so,cms-colord.so
```
[the remark this answers is not in the document]
Comment (EmbeddedAndroid) [reply meta-lmp-base/recipes-graphics/wayland/weston-init/weston.ini:4]: Other layers using weston.ini keep the commented out sections. I'm happy to trim them out though.

[the remark this answers is not in the document]
Comment (mike-scott) [reply meta-lmp-base/recipes-graphics/wayland/weston-init/weston.ini:4]: I'll defer to @ricardosalveti on this .. might be useful to have examples of the other config items if their useful.

File: meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend (hunk 2 above)
Comment (mike-scott) [reply meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend:4]: Maybe move this change to a file like: `meta-lmp-bsp/recipes-graphics/wayland/weston_%.bbappend` where we can land SoC specific config changes.

File: meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend (hunk 1 above)
Comment (mike-scott) [reply meta-lmp-base/recipes-graphics/wayland/weston_9.0.0.bbappend:8]: Maybe move this change to a file like: `meta-lmp-bsp/recipes-graphics/wayland/weston_%.bbappend` where we can land SoC specific config changes.  @ricardosalveti this isn't quite like a PACKAGECONFIG change like below.. what do you think?

**Comment by mike-scott** [comment]
Closing in favor of: https://github.com/foundriesio/meta-lmp/pull/275

**Pull request closed without merging**